### DISCUSSION TITLE
Docker container and build

### DISCUSSION THREAD

**Discussion by the-grue** [panix-os-dev-team/Panix#389] [General]
I'm interested in how you configure and build your docker container that performs the build process.  I haven't done any development on containers but am interested in how you have done this.  I used the container as part of the work I have done on the OS, but just took it for granted.  Now, if you are willing to share, I'd like to know more about this process.

Thanks!

Jim

**Comment by Kfeavel** [comment]
Hey @the-grue! The repo we have for the Docker files are at [Xyris/xyris-build](https://github.com/XyrisOS/xyris-build). Essentially you take a base image (like Ubuntu, Debian, Alpine, etc.) and then from there you essentially describe how you want to build the image. For us, it involves downloading the GCC and binutils source, extracting it, and the compiling it however best suites you.

As for the integration part, we use [JonathonReinhart/scuba](https://github.com/JonathonReinhart/scuba), which is tool that makes running commands in the docker container easier. The docs for it are available [here](https://scuba.readthedocs.io/en/latest/).

If you need any more help / have any more questions, let me know!

**Reply by the-grue** [reply]
Thanks @Kfeavel!  I'll give em a look.
